guvnor-structure/guvnor-structure-backend/src/test/java/org/guvnor/structure/backend/repositories/PullRequestServiceTest.java

```diff
@@ -0,0 +1,248 @@
+        PullRequest pullRequestA = service.createPullRequest( new PullRequestImpl( "child/a", "develop", "parent/a", "master" ) );
+        PullRequest pullRequestB = service.createPullRequest( new PullRequestImpl( "child/b", "develop", "parent/a", "develop" ) );
+        PullRequest pullRequestC = service.createPullRequest( new PullRequestImpl( "child/c", "develop", "parent/a", "master" ) );
+        PullRequest pullRequestD = service.createPullRequest( new PullRequestImpl( "child/d", "develop", "parent/a", "master" ) );
+
+        service.acceptPullRequest( pullRequestA.getTargetRepository(), pullRequestA.getId() );
+        service.rejectPullRequest( pullRequestB.getTargetRepository(), pullRequestB.getId() );
+        service.closePullRequest( pullRequestC.getTargetRepository(), pullRequestC.getId() );
+
+        final List<PullRequest> pullRequestsMerged = service.getAllPullRequestsFor( "parent/a", PullRequestStatus.MERGED );
+        final List<PullRequest> pullRequestsClosed = service.getAllPullRequestsFor( "parent/a", PullRequestStatus.CLOSED );
+        final List<PullRequest> pullRequestsOpened = service.getAllPullRequestsFor( "parent/a", PullRequestStatus.OPEN );
+        final List<PullRequest> pullRequestsRejected = service.getAllPullRequestsFor( "parent/a", PullRequestStatus.REJECTED );
+
+        assertEquals( 1, pullRequestsMerged.size() );
+        assertEquals( 1, pullRequestsClosed.size() );
+        assertEquals( 1, pullRequestsOpened.size() );
+        assertEquals( 1, pullRequestsRejected.size() );
+    }
+
+    @Test
+    public void testGeneratePullRequestId() {
+        PullRequest pullRequestOne = service.createPullRequest( new PullRequestImpl( "child/a", "develop", "parent/a", "master" ) );
+        PullRequest pullRequestTwo = service.createPullRequest( new PullRequestImpl( "child/a", "fix", "parent/a", "master" ) );
+        PullRequest pullRequestThree = service.createPullRequest( new PullRequestImpl( "child/a", "fix", "parent/a", "develop" ) );
+        PullRequest pullRequestFour = service.createPullRequest( new PullRequestImpl( "child/a", "fix", "parent/a", "master" ) );
+        assertEquals( 1, pullRequestOne.getId() );
+        assertEquals( 2, pullRequestTwo.getId() );
+        assertEquals( 3, pullRequestThree.getId() );
+        assertEquals( 4, pullRequestFour.getId() );
+    }
+
+    @Test
+    public void testBuildHiddenPath() {
+
+        final Path path = ( (PullRequestServiceImpl) service ).buildHiddenPath( 1, "source/a", "develop", "target/a", "master" );
+        final URI uri = URI.create( "git://PR-1-source/a/develop-master@target/a" );
+
+        verify( ioService ).get( eq( uri ) );
+    }
+
+    @Test
+    public void testBuildPath() {
+
+        final Path path = ( (PullRequestServiceImpl) service ).buildPath( "source/a", "develop" );
+        final URI uri = URI.create( "git://develop@source/a" );
+
+        verify( ioService ).get( uri );
+    }
+
+}
```

Comment:
I'm missing a test for binary files (ie. images).. how it would create conflicts.